README.md

```diff
@@ -4,6 +4,106 @@ An educational application to teach highschoolers various topics in cybersecurit
 
 ---
 
+## Adding a new module
+
+The process for adding a new module is documented here.
+
+Create a new application
+
+```shell
+./manage.py startapp example
+cd example
+rm tests.py admin.py models.py
+```
+
+Then edit the `apps.py` file to edit the module config class to inherit from `CyberEdAppConfig`.
+E.g., edit
+
+```python
+from django.apps import AppConfig
+
+class ExampleConfig(AppConfig):
+    name = 'example'
+```
+
+to look like this
+
+```python
+from shared.src.cyberedappconfig import CyberEdAppConfig
+
+class ExampleConfig(CyberEdAppConfig):
+    name = 'example'
+    module_name = 'An Example Application'
+    module_base_link = 'example/'
+    module_start_link = 'page1'
+    module_description = '...'
+```
+
+Then create a `urls.py` file and add the first page to the `urlpatterns` list.
+
+```python
+from django.urls import path
+from .apps import ExampleConfig
+from .views import ExampleBeginView
+
+urlpatterns = [path(ExampleConfig.module_start_link, ExampleBeginView.as_view(), name="page1")]
+```
+
+Finally, create a view and the corresponding HTML template for the begin page. Create a `views.py`
+file with the following contents.
+
+```python
+from django.views.generic import TemplateView
+
+class ExampleTemplateView(TemplateView):
+    def get_context_data(self, **kwargs):
+        """Add this module's name to the view's context."""
+        context = super().get_context_data(**kwargs)
+        context["module_name"] = HashingConfig.module_name
+        return context
+
+class ExampleBeginView(ExampleTemplateView):
+    """The main page for the hashing module."""
+    template_name = "example/page1.html"
+```
+
+Then create a templates directory and the the HTML template for the module begin page.
+
+```shell
+mkdir -p templates/example
+touch templates/example/page1.html
+```
+
+The extra directory with the same name as the application is used to prevent collisions between
```

Comment:
Is this the correct reasoning behind creating a (seemingly) unnecessary extra level of hierarchy?

Reply:
I'm not completely clear on which level you're talking about here; example?

Reply:
Why are the templates placed in `templates/example/page1.html` instead of `templates/page1.html`?

Reply:
Ah, yeah. Because as far as I can tell, django basically unions all the `templates` folders together, which means if the hashing module has `templates/page1.html` and the passwords module also has `templates/page1.html`, then there's no way to resolve them because the associated views are just linking to `page1.html`

Reply:
That's what I was guessing. Thanks for the confirmation.